src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -155,92 +156,93 @@ class JesBackend extends Backend with LazyLogging {
+          backendCall.call.task.outputs,
+          backendCall.callGcsPath,
+          backendCall.lookupFunction,
+          backendCall.engineFunctions
+        ).get
         Success(anonOutputs ++ namedOutputs)
     }
 
     // FIXME: Ignore all the errors!
     val unsafeJesOutputs: Seq[JesParameter] = jesOutputs.get
 
-    val jesParameters = standardParameters(callGcsPath) ++ jesInputs ++ unsafeJesOutputs
-
-    // Not possible to currently get stdout/stderr so redirect everything and hope the WDL isn't doing that too
-    JesConnection.storage.uploadObject(gcsExecPath, instantiatedCommandLine)
-
-    val run = Pipeline(s"/bin/bash exec.sh > $LocalStdoutValue 2> $LocalStderrValue", workflowDescriptor, call, jesParameters, GoogleProject, JesConnection).run
-    // Wait until the job starts (or completes/fails) before registering the abort to avoid awkward cancel-during-initialization behavior.
-    run.waitUntilRunningOrComplete()
-    abortFunctionRegistration.registrationFunction apply AbortFunction({() => run.abort()})
-    try {
-      val status = run.waitUntilComplete(None)
-
-      def taskOutputToRawValue(taskOutput: TaskOutput): Try[WdlValue] = {
-        val jesOutputFile = unsafeJesOutputs find { _.name == taskOutput.name } map { j => WdlFile(j.gcs) }
-        jesOutputFile.map(Success(_)).getOrElse(taskOutput.expression.evaluate(scopedLookupFunction, environment.engineFunctions, interpolateStrings = true))
-      }
-
-      def processOutput(taskOutput: TaskOutput, processingFunction: TaskOutput => Try[WdlValue]) = {
-        val rawValue = processingFunction(taskOutput)
-        logger.debug(s"JesBackend setting ${taskOutput.name} to $rawValue")
-        taskOutput.name -> rawValue
-      }
-
-      val outputMappings = call.task.outputs.map {taskOutput => processOutput(taskOutput, taskOutputToRawValue)}.toMap
-
-      lazy val stderrLength: BigInteger = JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(callGcsPath).gcs))
-
-      if (call.failOnStderr && stderrLength.intValue > 0) {
-        Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: stderr has length $stderrLength for command: $instantiatedCommandLine"))
-      } else status match {
-          case Run.Success =>
-            unwrapOutputValues(outputMappings, workflowDescriptor)
-          case Run.Failed(errorCode, errorMessage) =>
-            val throwable = if (errorMessage contains "Operation canceled at") {
-              new TaskAbortedException()
-            } else {
-              new Throwable(s"Workflow ${workflowDescriptor.id}: errorCode $errorCode for command: $instantiatedCommandLine. Message: $errorMessage")
-            }
-          Failure(throwable)
-      }
-    }
-    catch {
-      case e: Exception => Failure(e)
+    val jesParameters = standardParameters(backendCall.callGcsPath) ++ jesInputs ++ unsafeJesOutputs
+
+    backendCall.instantiateCommand match {
+      case Success(command) =>
```

Comment:
I would prefer to not have a 45 line closure hanging off of a Success. 

I realize though that this is largely c&p from the previous already too monolithic function. It'd be good for this to be broken down and I'd appreciate a stab being taken in that direction although it could be left for an eventual much needed refactoring of this entire file